docs/DeveloperGuide.md

```diff
@@ -1,34 +1,178 @@
+
+- [Introduction](#introduction)
+    * [To be added](#to-be-added)
+        + [To be added](#to-be-added)
+
+## 1. Introduction
+
+### 1.1 Software Overview
+
+easyLog is a Command Line Interface (CLI) application for warehouse employees to manage items and orders in the
+warehouse. easyLog allows users to add new items and orders, view existing items and orders, remove specific items and
+orders and clear all existing items and orders in the warehouse at ease. If the user types fast, easyLog can get
+logistic management tasks done faster than traditional Graphical User Interface (GUI) applications.
+
+easyLog, comprises a Parser component, Ui component, Storage component, Item component, Order component and Command
+component. Each component consists of various classes that work in tandem, to ensure it meets the purpose of our
+program.
+
+The purpose of this developer guide is to allow any interested contributors, who wish to develop this application
+further or just curious about the workings of this application. This would allow potential contributors to be able to
+dive right in to improving the code, performance, features or even adding new features much more easily due to the
+understanding of the structure of the codebase and implementation of existing features.
+
+### 1.2 Prerequisites
+
+- Java 11 (can be downloaded from [here](https://www.oracle.com/sg/java/technologies/javase-jdk11-downloads.html))
+- IntelliJ Idea (can be downloaded from [here](https://www.jetbrains.com/idea/download/#section=mac))
+
+### 1.3 Setting up
+
+1. Fork the easyLog repository [here](https://github.com/AY2021S2-CS2113T-T09-4/tp) and git clone it to any location on
+   your computer.
+
+2. Open any IDE (IntelliJ Idea preferred as mentioned in the prerequisites) and click `Configure` -> `Project Defaults`
+   -> `Project Structure` -> `New`
+
+3. Next, go to `Import Project` and select the build.gradle file.
+
+4. After opening the project, go to `src` -> `main` -> `java` -> `seedu.easylog` -> EasyLog and right click on the
+   EasyLog class.
+
+5. Upon successful run, the following opening message will be shown:
+
+```
+Hello! I'm easyLog!
+What can I do for you? Enter help to view commands.
+Looking for save data.
+Save data not found.
+```
+
+## 2. Design
+
+### 2.1 Architecture
+
+![Architecture Diagram](https://user-images.githubusercontent.com/57165946/112990632-ddf7e780-9198-11eb-99ec-f29aafbfc04f.png)
+
+
+The Architecture Diagram shown above illustrates the high-level design of easyLog. We will now proceed to explain each
+component and their respective functionalities below.
+
+### 2.2 EasyLog Component
+
+The EasyLog component initializes the app and looks for any available save file to load. It then proceeds to
+continuously take in input from the user until the user types in a exit command.
+
+### 2.3 Storage Component
+
+![Storage Diagram](https://user-images.githubusercontent.com/57165946/112991416-ab022380-9199-11eb-9ead-cdfcd19cf8e5.png)
```

Comment:
![image](https://user-images.githubusercontent.com/65816506/113094989-44c2e280-9225-11eb-80d5-607c3a185e17.png)

Should the top left box be omitted instead? (This can be seen in other UML diagrams in this DG as well)

For the Receipt class, should there be three rows instead? (Similar occurrences can be seen in other UML diagrams in this DG as well)